core/src/View/Kokkos_ViewTraits.hpp

```diff
@@ -228,25 +266,50 @@ template <class Traits>
 using accessor_from_view_traits_t =
     typename AccessorFromViewTraits<Traits>::type;
 
-template <class Traits, class Enabled = void>
-struct MDSpanViewTraits {
-  using mdspan_type = UnsupportedKokkosArrayLayout;
-};
-
 // "Natural" mdspan for a view if the View's ArrayLayout is supported.
-template <class Traits>
-struct MDSpanViewTraits<Traits, std::void_t<typename LayoutFromArrayLayout<
-                                    typename Traits::array_layout>::type>> {
+template <
+    class Traits,
+    class CustomizedArgs = decltype(customize_view_arguments(
+        ViewArguments<
+            typename Traits::value_type, typename Traits::array_layout,
+            typename Traits::device_type, typename Traits::memory_traits>())),
+    class Layout =
+        typename LayoutFromArrayLayout<typename Traits::array_layout>::type>
```

Comment:
Why do you need this template parameter? AFAICT, `MDSpanViewTraits` is only ever supposed to be used with one explicit template argument which means that `Layout` will always use the default template argument which doesn't depend on `CustomizedArgs`, right?

Reply:
Yeah but it depends on the Layout. Basically this is interoperability with Legacy View and custom layouts. Eventually this can go away. The trick used here is that `MDSpanViewTraits` can be instantiated even with a Layout that doesn't work for `mdspan` in which case it would produce a user error writing something about UnsupportedKokkosArrayLayout instead of some template error inside of mdspan, when the user would try to call `to_mdspan` on that legacy View.